### PR TITLE
Add/social instagram connection

### PR DESCRIPTION
**Pull request by gmjuhasz**
This change adds the Instagram connection to Calypso.

## Proposed Changes

* Added the new connection handlers
* Added service description and a new badge
* Added service example

## Testing Instructions

* Check out this PR
* Check the following change on wpcom: D111103-code. Sandbox the WP API.
* Go to `/marketing/connections/<your_site>` URL, and look for the new Instagram connection.
  * Description and looks should match with the designs
  * It should have a new badge
  * You should be able to connect an Instagram account, it should appear

The order comes from WPCOM, that needs to be changed there so we have instagram second

CleanShot 2023-05-23 at 10 59 19 png

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/marketing/connections/service-description.jsx [hunk 1]
```diff
@@ -33,6 +37,38 @@ class SharingServiceDescription extends Component {
...
+							'Share photos and video reels from your site to your Instagram Business account.',
+							{
+								comment: 'Description for Instagram Publicize when no accounts are connected',
+							}
+						) }
+						<InfoPopover className="instagram-business__info">
+							{ this.props.translate(
+								'Instagram requires a business account connected to Facebook in order to work with third party services.'
+							) }
+							<ExternalLink className="instagram-business__help-link" href="https://wordpress.org">
```
Comment (pablinos) [inline client/my-sites/marketing/connections/service-description.jsx]: Shall we change this to a Jetpack Redirect, so we don't have to create another PR to update it?

**Review by pablinos — APPROVED** [review]
This working well for me. I just wondered if we should use a Jetpack redirect for the support link.

File: client/my-sites/marketing/style.scss [hunk 2]
```diff
@@ -1144,3 +1150,13 @@
...
 }
+
+.service__new-badge.service__new-badge {
+	vertical-align: bottom;
+	background: #dfd1fb;
+	border-radius: 4px;
+	font-weight: 500;
+	font-size: $font-body-extra-small;
+	padding: 2px 10px;
+	margin-left: 5px;
```
Comment (manzoorwanijk) [inline client/my-sites/marketing/style.scss]: Can we use the [CSS Logical Properties](https://www.w3.org/TR/css-logical-1/) which are [RTL Aware](https://hacks.mozilla.org/2015/10/building-rtl-aware-web-apps-websites-part-2/)?

```suggestion
	margin-inline-start: 5px;
```

File: client/my-sites/marketing/connections/service-description.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.instagram-business {
+	&__info {
+		margin-left: 2px;
```
Comment (manzoorwanijk) [inline client/my-sites/marketing/connections/service-description.scss]:
```suggestion
		margin-inline-start: 2px;
```

File: client/my-sites/marketing/connections/service-description.jsx (hunk 1 above)
Comment (manzoorwanijk) [reply client/my-sites/marketing/connections/service-description.jsx]: I have added a Jetpack redirect.

File: client/my-sites/marketing/connections/service-examples.jsx [hunk 4]
```diff
@@ -181,6 +184,50 @@ class SharingServiceExamples extends Component {
...
+					"{{italic}}When you click “connect” you'll be asked to {{strong}}log into Facebook{{/strong}}. If your Instagram Business account isn't listed, ensure it's linked to a Facebook page.{{/italic}}",
+					{
+						components: {
+							strong: <strong />,
+							italic: <i />,
+						},
+					}
+				) }
+				<br />
+				<ExternalLink className="instagram-business__help-link" href="https://wordpress.org">
```
Comment (pablinos) [inline client/my-sites/marketing/connections/service-examples.jsx]: Sorry, @manzoorwanijk, there's this one too!

**Review by pablinos — COMMENTED** [review]
Looks good, there's just another link to update.

File: client/my-sites/marketing/connections/service-examples.jsx (hunk 4 above)
Comment (manzoorwanijk) [reply client/my-sites/marketing/connections/service-examples.jsx]: Sorry for missing that one. It's done now. Thanks.

**Review by pablinos — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8009924

Thank you @gmjuhasz for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8009924

Thank you @gmjuhasz for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8009924).

**Pull request merged**